lightning/src/ln/channel.rs

```diff
@@ -7054,10 +7089,10 @@ impl<SP: Deref> FundedChannel<SP> where
 						//   MUST send its tx_signatures for that funding transaction.
 						// else if it HAS received commitment_signed AND has received tx_signatures for that funding transaction:
 						//   MUST send its tx_signatures for that funding transaction.
-						let tx_signatures = if session.has_received_commitment_signed() && ((
-							!session.counterparty_sent_tx_signatures() &&
+						let tx_signatures = if (our_commitment_transaction == INITIAL_COMMITMENT_NUMBER - 1) && ((
```

Comment:
Wait, this and below are dual funding-specific as they assume initial commitment. I'll need to make it work in general.